pqarrow/builder/optbuilders.go

```diff
@@ -637,6 +638,22 @@ func (b *OptInt32Builder) Set(i int, v int32) {
 	b.data[i] = v
 }
 
+// UnsafeSwap swaps without bounds check.
+func (b *OptInt32Builder) UnsafeSwap(i, j int) {
+	b.data[i], b.data[j] = b.data[j], b.data[i]
+}
+
+// UnsafeSet sets v at index i without bounds check.
+func (b *OptInt32Builder) UnsafeSet(i int, v int32) {
+	b.data[i] = v
+	bitutil.SetBit(b.validityBitmap, i)
```

Comment:
Interesting, is the fact that we don't set the null bitmap in `Set` a bug? Same idea here as below in `UnsafeValue`: I don't think the application-level bounds check is necessary so we could probably merge